draft-ietf-quic-recovery.md

```diff
@@ -1608,17 +1615,20 @@ Invoked when DetectAndRemoveLostPackets deems packets lost.
 
 ~~~
    InPersistentCongestion(largest_lost):
-     // Persistent congestion cannot be declared on the
-     // first RTT sample.
-     if (is first RTT sample):
-       return false
      pto = smoothed_rtt + max(4 * rttvar, kGranularity) +
        max_ack_delay
-     congestion_period = pto * kPersistentCongestionThreshold
+     pc_period = pto * kPersistentCongestionThreshold
+
+     // Only look for persistent congestion if the period
+     // starts after getting the first RTT sample.
+     assert(first_rtt_sample is not never)
+     if (largest_lost.time_sent - pc_period > first_rtt_sample):
```

Comment:
```suggestion
     if (smallest_lost.time_sent - pc_period > first_rtt_sample):
```

Reply:
`largest_lost` is the name of the variable we have.  This says that the span of length pc_period that ends at the time that the largest lost was sent must start after we first got an RTT sample.

Reply:
FWIW, this is AllPacketsLost is not a fair representation of the algorithm in the latest PR.  Only the largest ack-eliciting packet is relevant, and the function needs to ensure that it only considers the first ack-eliciting packet.  But I guess we're abstracting over much of that.

Reply:
Yes, we're glossing over a few implementation details with AreAllPacketsLost() below, but I think this is detailed enough.

Reply:
See my comment below. You can change the param here to lost_packets. I would suggest the following:

We could make `AreAllPacketsLostInPeriod()` return the starting point of the persistent congestion period, and use that up here. Meaning call that function first and then check to see if that period starts after first_rtt_sample.